src/config/backpack/crud.php

```diff
@@ -62,6 +62,10 @@
             // thus clearing any searching, filtering or pagination that has been
             // remembered and persisted using persistentTable
             'resetButton' => true,
+
+            // Hide list action button labels keeping only the icon
+            // Labels occupy some space on the table and icons may be enough in some usecases
+            'hideTableButtonsText' => false,
```

Comment:
Maybe 
```php
'defaultButtonTextClass' => [
    'top' => 'd-none',
    'line' => 'd-none', // alternative: 'd-sm-none d-md-none d-lg-inline'
    'bottom' => 'd-none',
];

// afterwards in the buttons used as:
config('backpack.crud.list.buttonTextClass.top')
```

Don't know. REALLY don't know. Let's think about it.

Reply:
@pxpm what do you think?
I think it's not straight forward, but it's a "wider" solution.

Reply:
I agree with that granularity, but still don't address my issue. I don't want my buttons to have labels at all, nor in mobile or tablet or desktop. I just want icons, can I get that ? 

No pun intended here, I'm just an unsatisfiable developer with reasonable requests 🐰 

In future I might want some buttons with icons, other don't. And pretty sure that If we could setup labels under the icon instead of inline .... 

These last two are reasonable too but out of scope, i'am just messing around. If I can get icons without text on every screen size I think I would settle with further requests for a while.

Reply:
@pxpm you can by using bootstrap classes;

```php
'defaultButtonTextClass' => [
    'top' => '',
    'line' => 'd-none',
    'bottom' => 'd-sm-none d-md-none d-lg-inline',
];
```

For instance, this would;
- not touch top buttons
- hide all labels on inline table buttons
- hide for mobile `sm` and tablet `md`, and show it on desktop `lg` for bottom buttons

Maybe it's missing me something, or maybe you missed it 😅 as I said **it's not straight forward,** and that is kind of a problem 😕

Reply:
Guys... sorry to burst the bubble here. But I have one more idea to change the scope of this feature, that I think will address @pxpm 's valid concerns too. Hear me out, trying to make a summary of the process that's happened so far too:

- **the initial idea** was to add `span`s so that the dev can hide the labels using CSS; 
     - PROs: very small change; allows dev to show/hide stuff using CSS, which is natural;
     - CONs: no granularity, or very difficult to achieve by using custom CSS selectors;
- **the second idea** was to make the 80% use case even easier, by writing the CSS inside a pre-defined config value;
     - PROs: easier to use for the 80% use case;
     - CONs: bigger change;
- **third idea** now: why don't we use the `CrudButton` object, add a few properties there, which will be reflected in the blade file that gets loaded by that class? 
     - PROs: 
         - the user can change the CSS from the config for all CRUDs, but also inside a CrudController individually for one particular button; using the feature would be very similar to `wrapperAttributes` and `attributes` (aka change whatever you want);
     - CONs: even bigger change; complicates all buttons for ever;

Pseudocode of how this third idea could look like inside a button:
```php
{{-- BEFORE --}}

@if ($crud->hasAccess('create'))
	<a href="{{ url($crud->route.'/create') }}" 
		class="btn btn-primary">
		<i class="la la-plus"></i> {{ trans('backpack::crud.add') }} {{ $crud->entity_name }}
	</a>
@endif

{{-- AFTER --}}

@php
$button->attributes['class'] = $button->attributes['class'] ?? 'btn btn-primary';
$button->iconAttributes['class'] = $button->iconAttributes['class'] ?? 'la la-plus';
$button->spanAttributes['class'] = $button->spanAttributes['class'] ??
@endphp

@if ($crud->hasAccess('create'))
	<a href="{{ url($crud->route.'/create') }}"
		@include('crud::buttons.inc.attributes')
		>
		<i class="{{ $button->iconAttributes['class'] }}"></i> <span class="{{ $button->spanAttributes['class'] }}">{{ trans('backpack::crud.add') }} {{ $crud->entity_name }}</span>
	</a>
@endif
```

Which brings me to ask - **Is this really worth it?!** Is this feature worth changing all our buttons, and making them complicated, when it's so _so_ simple to just overwrite them? I think we've now generated ideas for this at the two opposite sides of the spectrum:
- minimally invasive (just add spans)
- maximally invasive (but also maximum configurability)

So I think we should return to a minimally-invasive midset, and do this:

1. (aka **Fourth Idea** ) Let's create another PR that just adds `span`s, merge that PR into 4.1 and move this PR to 4.2. The 4.1 PR would provide a way to do this, using CSS in a CSS file, for those who need it, while being minimally invasive (_KISS - Keep It Simple Stupid_). To increase granularity, we could also add a `data-attribute` on the button, with the button name. So that it's dead-simple to select what button you want using good-old CSS selectors (`a[data-name=create] span { display: none }`). This can be merged into 4.1 if you think this FEATURE (!!! 🤣 ) is worth supporting before 4.2.

2. Let's continue the conversation here, about the best way to do this, but keep in mind that enlarging the scope of the change would NOT be part of 4.1 but 4.2. So it's not a priority.

Personally I think doing after we implement step 1 above, we'll be happy enough like that, and won't need to do anything else. If we do need to add on top of it, it'll be easy to keep the backwards-compatibility because the `data-name` and `span` will be needed in the extended version too. Correct me if I'm wrong.

The fourth idea does have one major downside - that you'll be forced to do things using CSS, not bootstrap classes. But as far as I can tell... that's pretty much it.

Thoughts?